src/main/java/htsjdk/samtools/SAMFileWriterFactory.java

```diff
@@ -372,32 +374,35 @@ private SAMFileWriter initWriter(final SAMFileHeader header, final boolean preso
      */
     public SAMFileWriter makeSAMOrBAMWriter(final SAMFileHeader header, final boolean presorted, final File outputFile) {
         final String filename = outputFile.getName();
-        if (filename.endsWith(BamFileIoUtils.BAM_FILE_EXTENSION)) {
-            return makeBAMWriter(header, presorted, outputFile);
-        }
-        if (filename.endsWith(".sam")) {
+
+        if (SAM_TYPE.hasValidFileExtension(filename)) {
             return makeSAMWriter(header, presorted, outputFile);
+        } else {
+            if (!BAM_TYPE.hasValidFileExtension(filename)) {
+                log.warn("Unknown file extension, assuming BAM format when writing file: " + outputFile.getAbsolutePath());
+            }
```

Comment:
As a user I'd prefer to be warned because defaulting to BAM is not obvious. We don't write SRA format out for example, so silent errors are possible.